dali/pipeline/pipeline.h

```diff
@@ -132,15 +132,15 @@ class DLL_PUBLIC Pipeline {
 
 
   template<typename T, typename OperatorBackend>
-  void
-  SetDataSourceHelper(const string &name, const T &tl, OperatorBase *op_ptr, AccessOrder order = {},
-                      InputOperatorSettingMode in_op_setting_mode = {}) {
+  void SetDataSourceHelper(const string &name, const T &tl, std::optional<std::string> data_id,
```

Comment:
Can you swap the order of template arguments? This way there will be no need to specify `T` as a template argument.